### PR TITLE
Add MultiVCFAnalyzer

### LINKED ISSUE #852: new module: multivcfanalyzer
I think it would be good to have a module for [multivcfanalyzer](https://bioconda.github.io/recipes/multivcfanalyzer/README.html)

- [x] This module does not exist yet with the [`nf-core modules list`](https://github.com/nf-core/tools#list-modules) command
- [x] There is no [open pull request](https://github.com/nf-core/modules/pulls) for this module
- [x] There is no [open issue](https://github.com/nf-core/modules/issues) for this module
- [x] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by jfy133**
## PR checklist

Closes #852 

Replaces #996 which is highly out of date in terms of nf-core module structure

Note: $args is not used because the input channels are strictly order based and any interfereance can break the tool 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/multivcfanalyzer/main.nf [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    conda (params.enable_conda ? "bioconda::multivcfanalyzer=0.85.2" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/multivcfanalyzer:0.85.2--hdfd78af_1':
+        'quay.io/biocontainers/multivcfanalyzer:0.85.2--hdfd78af_1' }"
+
+    input:
+    path vcfs
+    path fasta
+    path snpeff_results
+    path gff
```
Comment (maxulysse) [inline modules/multivcfanalyzer/main.nf:14]: Don't you want a single tuple for the input, and separate path for referecnes?

**Comment by maxulysse** [comment]
ORDER BASED?

File: modules/multivcfanalyzer/main.nf [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    path('structureGenotypes_noMissingData-Columns.tsv') , emit: structure_genotypes_nomissing
+    path('MultiVCFAnalyzer.json')                        , emit: json
+    path "versions.yml"                                  , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def args2 = task.ext.args2 ?: ''
```
Comment (maxulysse) [inline modules/multivcfanalyzer/main.nf:42]: Why not removing that if you don't use it?
Is it necessary for linting?

File: modules/multivcfanalyzer/main.nf (hunk 1 above)
Comment (jfy133) [reply modules/multivcfanalyzer/main.nf:14]: >Don't you want a single tuple for the input, and separate path for referecnes?

I don't follow, sorry. Could you rephrase? There is a separate path for fasta here?

File: modules/multivcfanalyzer/main.nf (hunk 2 above)
Comment (jfy133) [reply modules/multivcfanalyzer/main.nf:42]: Consistency.

I assumed it's better to call `args2` to indicate that it's for the second part of the module (gzipping). I could remove the args (1) line entirely. I don't want to confuse a future reviewer of the module wondering where args1 was.

I also worry someone would try adding it to `modules.conf` (even if it does nothing) and it could cause errors? Although maybe not thinking about it :thinking:  

Don't mind ultaimtely - what do you think?

File: modules/multivcfanalyzer/main.nf (hunk 1 above)
Comment (maxulysse) [reply modules/multivcfanalyzer/main.nf:14]: I'm assuming that vcfs and snpeff_results are your input, and gff and fasta your references.
so why not:
```
tuple path(vcfs), (snpeff_results)
path fasta
path gff
```

Comment (jfy133) [reply modules/multivcfanalyzer/main.nf:14]: ah I see. No - it's a single snpEff file as input.

Comment (jfy133) [reply modules/multivcfanalyzer/main.nf:14]: The real input is the vcfs though, and then other reference files (fasta, gff)...?

It is very confusing... main input are the VCFs of multiple files, but you can include the annotation files based on the reference you used...

File: modules/multivcfanalyzer/main.nf (hunk 2 above)
Comment (jfy133) [inline modules/multivcfanalyzer/main.nf:42]:
```suggestion
    // def args = task.ext.args ?: '' // MultiVCFAnalyzer has strict and input ordering and all are mandatory. Deactivate $args to prevent breakage of input 
    def args2 = task.ext.args2 ?: ''
```

**Comment by maxulysse** [comment]
Any idea why the checksums are different with Singularity?
Anyhow tests are ok with Docker, so LGTM

**Comment by jfy133** [comment]
> Any idea why the checksums are different with Singularity? Anyhow tests are ok with Docker, so LGTM

No clue, it wasn't happening before :\ I will check tomorrow or monday

**Comment by jfy133** [comment]
Thanks for the suggestiosn @rpetit3! Indeed that solved the md5 sum stuff (I suspect it's to do with time stamps, but there wasn't a `--nonname` option in these particular containers). 

@maxulysse looks ok now?

**Review by maxulysse — APPROVED** [review]
:100:
